ably/helpers_test.go

```diff
@@ -18,6 +18,93 @@ func (opts *ClientOptions) RealtimeURL() string {
 	return opts.realtimeURL()
 }
 
+// StateRecorder provides:
+//
+//   * send ably.State channel for recording state transitions
+//   * goroutine-safe access to recorded state enums
+//
+type StateRecorder struct {
+	ch     chan State
+	states []StateEnum
+	wg     sync.WaitGroup
+	done   chan struct{}
+	mtx    sync.Mutex
+}
+
+func NewStateRecorder() *StateRecorder {
+	return newRecorderBuffer(16)
+}
+
+func newRecorderBuffer(buffer int) *StateRecorder {
```

Comment:
I'm not sure why this needs to be its own function. It would probably be as good to have this directly in NewStateRecorder.